prime-router/src/main/kotlin/fhirengine/engine/FHIRDestinationFilter.kt

```diff
@@ -0,0 +1,279 @@
+                }
+                fhirEngineRunResults(message, actionHistory)
+            }
+
+            else -> {
+                throw RuntimeException(
+                    "Message was not a FhirDestinationFilter and cannot be processed: $message"
+                )
+            }
+        }
+    }
+
+    /**
+     * Process a [queueMessage] off of the raw-elr azure queue, convert it into FHIR, and store for next step.
+     * [actionHistory] ensures all activities are logged.
+     */
+    private fun fhirEngineRunResults(
+        queueMessage: ReportPipelineMessage,
+        actionHistory: ActionHistory,
+    ): List<FHIREngineRunResult> {
+        val contextMap = mapOf(
+            MDCUtils.MDCProperty.ACTION_NAME to actionHistory.action.actionName.name,
+            MDCUtils.MDCProperty.REPORT_ID to queueMessage.reportId,
+            MDCUtils.MDCProperty.TOPIC to queueMessage.topic,
+            MDCUtils.MDCProperty.BLOB_URL to queueMessage.blobURL
+        )
+        withLoggingContext(contextMap) {
+            // track input report
+            logger.info("Starting FHIR DestinationFilter step")
+            actionHistory.trackExistingInputReport(queueMessage.reportId)
+
+            // pull fhir document and parse FHIR document
+            val fhirJson = LogMeasuredTime.measureAndLogDurationWithReturnedValue(
+                "Downloaded content from queue message"
+            ) {
+                queueMessage.downloadContent()
+            }
+            val bundle = FhirTranscoder.decode(fhirJson)
+            val bodyString = FhirTranscoder.encode(bundle)
+
+            // go up the report lineage to get the sender of the root report
+            val sender = reportService.getSenderName(queueMessage.reportId)
+
+            // send event to Azure AppInsights
+            val observationSummary = AzureEventUtils.getObservations(bundle)
+            azureEventService.trackEvent(
+                ReportAcceptedEvent(
+                    queueMessage.reportId,
+                    queueMessage.topic,
+                    sender,
+                    observationSummary,
+                    fhirJson.length
+                )
+            )
+
+            // get the receivers that this bundle should go to
+            val receivers = findTopicReceivers(queueMessage.topic).filter { receiver ->
+                receiver.jurisdictionalFilter.all { filter ->
+                    FhirPathUtils.evaluateCondition(
+                        CustomContext(bundle, bundle, shorthandLookupTable, CustomFhirPathFunctions()),
+                        bundle,
+                        bundle,
+                        bundle,
+                        filter
+                    )
+                }.also { pass ->
+                    if (!pass) {
+                        azureEventService.trackEvent(
+                            DestinationFilterReportNotRoutedEvent(
```

Comment:
Looking at this again, I lean slightly towards this just getting logged or not doing anything at all; this will be very noisy otherwise since we'll fire this for every receiver not getting the message.